src/main/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/steps/ProvideLogsStep.groovy

```diff
@@ -0,0 +1,230 @@
+                    artifactsMap.put(relativePath, relativePath)
+                }
+                run.artifactManager.archive(workspace, launcher, listener, artifactsMap)
+                run.addAction(new ProvideLogsAction(run))
+                workspace.child(logDirName).deleteContents()
+                listener.logger.println("Successfully added ecu.test logs to jenkins.")
+            }
+            listener.logger.flush()
+        }
+    }
+
+    private static final class ExecutionCallable extends MasterToSlaveCallable<Void, IOException> {
+
+        private static final long serialVersionUID = 1L
+
+        private final long startTimeMillis
+        private final EnvVars envVars
+        private final String logDirPath
+        private final TaskListener listener
+
+
+        ExecutionCallable(long startTimeMillis, EnvVars envVars, String logDirPath, TaskListener listener) {
+            super()
+            this.startTimeMillis = startTimeMillis
+            this.envVars = envVars
+            this.logDirPath = logDirPath
+            this.listener = listener
+        }
+        /**
+         * Calls downloadReportFolder via the RestApiClient for all report ids.
+         * Then extracts and saves the ecu.test log out of the returned zip folder.
+         * Logs a warning if no reports were returned by ecu.test.
+         * @return List of strings containing the paths of the logs.
+         */
+        @Override
+        Void call() throws IOException {
+            listener.logger.println("Providing ecu.test logs to jenkins.")
+            try {
+                RestApiClient apiClient = RestApiClientFactory.getRestApiClient(envVars.get('ET_API_HOSTNAME'), envVars.get('ET_API_PORT'))
+                List<ReportInfo> reports = apiClient.getAllReports()
+
+                if (reports == null || reports.isEmpty()) {
+                    listener.logger.println("[WARNING] No report files returned by ecu.test")
+                }
+                for (def report : reports) {
+                    String reportDir = report.reportDir.split('/').last()
+                    if (!checkReportFolderCreationDate(reportDir)) {
+                        listener.logger.println("[WARNING] ecu.test reports contains folder older than this run. Path: ${report.reportDir}")
+                    }
+
+                    File reportFolderZip = apiClient.downloadReportFolder(report.testReportId)
+                    if (!extractAndSaveFromZip(reportFolderZip, ["ecu.test_out", "ecu.test_err"], "${logDirPath}/${reportDir}")) {
+                        throw new AbortException("${report.reportDir} is missing one or all log files!")
+                    }
+                }
+            }
+            catch (ApiException e) {
+                if (e instanceof de.tracetronic.cxs.generated.et.client.v2.ApiException && e.code == 404) {
+                    throw new AbortException("[ERROR] Downloading reportFolder is not available for ecu.test < 2024.2!")
```

Comment:
different error message. Maybe store that within a variable like `unsupportedProvideLogsMessage`